carbon-route.html

```diff
@@ -271,48 +276,73 @@
         tailPath = '/' + tailPath;
       }
 
-      this._skipMatch = true;
       this._matched = matched;
-      this.data = namedMatches;
+
       var tailPrefix = this.route.prefix + matched;
 
-      if (!this.tail || this.tail.prefix !== tailPrefix || this.tail.path !== tailPath) {
-        this.tail = {
+
+      // HACK(rictic): skirting around 1.0's lack of a setMulti by poking at
+      //     internal data structures. I would not advise that you copy this
+      //     example.
+      //
+      //     This is juggling footguns.
+      this._propertySetter('data', namedMatches);
+
+      var tailChanged =
+          !this.tail ||
+          this.tail.prefix !== tailPrefix ||
+          this.tail.path !== tailPath;
+      if (tailChanged) {
+        this._propertySetter('tail', {
           prefix: tailPrefix,
           path: tailPath,
           __queryParams: this.route.__queryParams
-        };
+        });
+      }
+
+      this._dataInUrl = {};
+      for (var key in namedMatches) {
+        this._dataInUrl[key] = namedMatches[key];
       }
+
       this._setActive(true);
-      this._skipMatch = false;
+      if (tailChanged) {
+        this._pathEffector('tail', this.tail);
+        this._notifyPathUp('tail', this.tail);
+      }
+      this._pathEffector('data', this.data);
+      this._notifyPathUp('data', this.data);
```

Comment:
@timvdlippe Yes, absolutely. The core team is aware of the desire for such a function, but will not be implementing it for the 1.0 lineage of Polymer core. There are plans to implement a more wholesome implementation in the 2.0 lineage. In the mean time, this helper would be a stop-gap (also please see `HACK` warning above).